src/axom/quest/SignedDistance.hpp

```diff
@@ -33,35 +35,52 @@ namespace axom
 {
 namespace quest
 {
-template <int NDIMS>
+namespace detail
+{
+struct UcdMeshData
+{
+  int shape_type;
+  mint::CellType single_cell_type;
+  const mint::CellType* cell_types;
+  const IndexType* cells_to_nodes;
+  IndexType nodes_per_cell;
+  const IndexType* cell_node_offsets;
+
+  AXOM_HOST_DEVICE mint::CellType getCellType(IndexType cellId) const;
+
+  AXOM_HOST_DEVICE const IndexType* getCellNodeIDs(IndexType cellId,
+                                                   int& nnodes) const;
+};
```

Comment:
Yeah, this essentially pulls the data pointers for the unstructured mesh connectivity from a polymorphic `mint::Mesh` object into a POD-like struct, so that we can use mesh data on the GPU.